From 2a8f6fbd54eae37c4689cdc630c5f98782733e1b Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Tue, 22 Apr 2014 13:25:16 +0200
Subject: [PATCH 04/28] Cleaner: Remove the database file on disabling

Instead of manually cleaning every file which would consume a lot of
time and cpu.
---
 src/file/cleaner/main.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/file/cleaner/main.cpp b/src/file/cleaner/main.cpp
index fa08071..dd26f88 100644
--- a/src/file/cleaner/main.cpp
+++ b/src/file/cleaner/main.cpp
@@ -29,9 +29,12 @@
 #include <KComponentData>
 #include <QApplication>
 #include <QDBusConnection>
+#include <QFile>
+#include <QDir>
 
 #include <KDebug>
 #include <KStandardDirs>
+#include <KConfigGroup>
 
 int main(int argc, char* argv[])
 {
@@ -58,6 +61,18 @@ int main(int argc, char* argv[])
 
     const QString path = KGlobal::dirs()->localxdgdatadir() + "baloo/file/";
 
+    KConfig config("baloofilerc");
+    KConfigGroup group = config.group("Basic Settings");
+    bool indexingEnabled = group.readEntry("Indexing-Enabled", true);
+    if (!indexingEnabled) {
+        QDir dir(path);
+        Q_FOREACH (const QString& file, dir.entryList(QDir::Files)) {
+            dir.remove(file);
+        }
+        QFile::remove(path);
+        return 0;
+    }
+
     Database db;
     db.setPath(path);
     db.init();
-- 
1.9.0

